src/test/java/net/masterthought/cucumber/ReportBuilderTest.java

```diff
@@ -62,6 +62,23 @@ void ReportBuilder_storesFilesAndConfiguration() {
         assertThat(assignedConfiguration).isSameAs(configuration);
     }
 
+    @Test
+    void ReportBuilder_setsAndGetsCustomReportParser(){
+        // given
+        final List<String> jsonFiles = new ArrayList<>();
+        final Configuration configuration = new Configuration(null, null);
+        final ReportParser reportParser = new ReportParser(configuration);
+
+        // when
+        ReportBuilder builder = new ReportBuilder(jsonFiles, configuration);
+        builder.setReportParser(reportParser);
+
+        // then
+        ReportParser assignedReportParser = Whitebox.getInternalState(builder, "reportParser");
+
+        assertThat(assignedReportParser).isSameAs(builder.getReportParser());
```

Comment:
could be just
`assertThat(reportParser).isSameAs(builder.getReportParser());` and no need to use `Whitebox`

Reply:
Duly noted Damian! Thank you.